cmd/ethrex/utils.rs

```diff
@@ -81,9 +82,20 @@ pub fn read_block_file(block_file_path: &str) -> Block {
         .unwrap_or_else(|_| panic!("Failed to decode block file {}", block_file_path))
 }
 
-pub fn read_genesis_file(genesis_file_path: &Path) -> Genesis {
+pub fn read_genesis_file(genesis_file_path: &Path) -> Result<Genesis, ChainError> {
```

Comment:
the function is called `read_genesis_file`, and this function is doing more than that. Maybe you should create Genesis::new or Genesis::from